src/utils/__tests__/use-previous.test.ts

```diff
@@ -0,0 +1,47 @@
+import {renderHook, act} from '@testing-library/react-hooks';
+import {usePrevious} from '../use-previous';
+
+describe('usePrevious', () => {
+  it('should return undefined on initial render', () => {
+    const {result} = renderHook(() => usePrevious(0));
+    expect(result.current).toBeUndefined();
+  });
+
+  it('should return the previous value after update', () => {
+    let value = 0;
+    const {result, rerender} = renderHook(() => usePrevious(value));
+
+    act(() => {
+      value = 1;
+      rerender();
+    });
+
+    expect(result.current).toBe(0);
+
+    act(() => {
+      value = 2;
+      rerender();
+    });
+
+    expect(result.current).toBe(1);
+  });
+
+  it('should not update previous value if current value does not change', () => {
+    let value = {a: 1};
+    const {result, rerender} = renderHook(() => usePrevious(value));
+
+    act(() => {
+      value = {a: 1}; // new object with the same content
+      rerender();
+    });
+
+    expect(result.current).toBeUndefined();
+
+    act(() => {
+      value = {a: 2}; // change content
+      rerender();
+    });
+
+    expect(result.current).toEqual({a: 1});
+  });
+});
```

Comment:
`usePrevious` is a pretty well-established custom hook in the community, and this is not inline with how it normally works. As of my understanding `usePrevious` is used for accessing the previous props/state that was used in a render, much like `prevProps` in `componentDidUpdate`.

I think the hook should have a different name than `usePrevious` if it should work like this.

Reply:
Updated: `useStableProp` has been renamed to `useStableValue`, and `usePrevious` has been renamed to `useStablePreviousValue`